### PR TITLE
Feature/update delete

### PR DESCRIPTION
**Pull request by Raj-kadevar**

### CODE REVIEW

File: user/forms.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
+from django import forms
+from user.models import UserEmployee
+
+
+class Registration(forms.ModelForm):
```
Comment (yatishTrootech) [inline user/forms.py]: RegistrationForm

File: user/views.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+class UserView(CreateView):
+    def get_queryset(self):
+        return UserEmployee.objects.all()
+
+    def get_context_data(self, **kwargs):
+        form = Registration
+        return {"form": form}
+
+    def get_template_names(self):
```
Comment (yatishTrootech) [inline user/views.py]: why do we need to override this ?

File: user/views.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        return "user/form.html"
+
+    def post(self, request, *args, **kwargs):
+        user = Registration(request.POST)
```
Comment (yatishTrootech) [inline user/views.py]: RegistrationForm

File: user/views.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+        if user.is_valid():
+            user_data = user.save()
+            user_data.set_password(user_data.password)
+            user_data.save()
+        else:
+            form = Registration()
```
Comment (yatishTrootech) [inline user/views.py]: where did you use form variable ?

File: user/views.py [hunk 5]
```diff
@@ -0,0 +1,57 @@
+from django.http import HttpResponseRedirect
+from django.shortcuts import render, redirect
+from django.views.generic import CreateView
+from user.forms import Registration
+from user.models import UserEmployee
+
+
+class UserView(CreateView):
+    def get_queryset(self):
```
Comment (yatishTrootech) [inline user/views.py]: Why this method is overrided in create view

File: user/views.py [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+        else:
+            form = Registration()
+            error = user.errors
+            return render(request, "user/form.html", {"err": error, "form": user})
+        return HttpResponseRedirect("index")
+
+
+def index(request):
+    data = UserEmployee.objects.all()
+    return render(request, "user/index.html", {"obj": data})
```
Comment (yatishTrootech) [inline user/views.py]: obj name should be users

Comment (yatishTrootech) [reply user/views.py]: err ane should be errors

File: user/views.py [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+
+
+def delete(request, id):
+    data = UserEmployee.objects.get(id=id)
+    data.delete()
+    return redirect("index")
```
Comment (yatishTrootech) [inline user/views.py:38]: What if user id does not exists ?

File: user/views.py [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+
+
+def update(request, id):
+    get_user = UserEmployee.objects.get(id=id)
+    if request.method == "POST":
+        name = request.POST["name"]
```
Comment (yatishTrootech) [inline user/views.py]: Why all fields are manually updated it should be done using form

File: user/views.py [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+        phone_num = request.POST["phone_number"]
+        address = request.POST["address"]
+        get_user = UserEmployee.objects.get(id=id)
+        get_user.username = name
+        get_user.email = email
+        get_user.phone_number = phone_num
+        get_user.address = address
+        get_user.save()
+        return redirect("index")
+    return render(request, "user/update.html", {"key": get_user})
```
Comment (yatishTrootech) [inline user/views.py]: what is key ? use proper name

**Review by yatishTrootech — CHANGES_REQUESTED** [review]

File: user/views.py [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+        get_user = UserEmployee.objects.get(id=id)
+    except ObjectDoesNotExist as e:
+        return redirect("index")
+
+    if request.method == "POST":
+        user = UpdateForm(request.POST, instance=get_user)
+        if user.is_valid():
+            user.save()
+            return redirect("index")
+    return render(request, "user/update.html", {"user_information": get_user})
```
Comment (yatishTT) [inline user/views.py:52]: user: user

File: user/views.py [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+
+def delete(request, id):
+    data = UserEmployee.objects.get(id=id)
+    data.delete()
+    return redirect("index")
+
+
+def update(request, id):
+    try:
+        get_user = UserEmployee.objects.get(id=id)
```
Comment (yatishTT) [inline user/views.py:43]: user

File: user/views.py [hunk 12]
```diff
@@ -0,0 +1,49 @@
...
+    except ObjectDoesNotExist as e:
```
Comment (yatishTT) [inline user/views.py:44]: user.DoesNotExis 

error msg = "user with this id does not exits"

**Review by yatishTT — CHANGES_REQUESTED** [review]

**Review by yatishTT — APPROVED** [review]

**Pull request merged**